feat: add isexpand and complete_match (#16716)
vim/vim: runtime/doc/options.txt

	and there is a letter before it, the completed part is made uppercase.
	With 'noinfercase' the match is used as-is.

						*'isexpand'* *'ise'*
'isexpand' 'ise'	string	(default: "")
			local to buffer

> 	A comma-separated list of defines characters for expand completion in

Reviewer: ```suggestion
	A comma-separated list of characters for triggering completion in
```

	and there is a letter before it, the completed part is made uppercase.
	With 'noinfercase' the match is used as-is.

						*'isexpand'* *'ise'*
'isexpand' 'ise'	string	(default: "")
			local to buffer
	Defines characters and patterns for completion in insert mode. Used by
	the |complete_match()| function to determine the starting position for
	completion. This is a comma-separated list of triggers. Each trigger
	can be:
	- A single character like "." or "/"

> 	- A sequence of characters like "->", "/*", or "/**"

Reviewer 2: Should there be a way to escape `,`, so that `,` can be used in trigger chars?

Author: makes sense. Should we also support deduplication?

Author: also  careful consideration, it should return a nested list, such as `ise=abc,c`  return `[ [0, 'abc'], [2, 'c']`

Author: added by using `\\,`

Reviewer 2: Will it be better to use `copy_option_part()` to separate triggers?

Author: yep should be better

Reviewer 2: Actually I was talking about the part in `f_complete_match()`

Author: updated

	and there is a letter before it, the completed part is made uppercase.
	With 'noinfercase' the match is used as-is.

						*'isexpand'* *'ise'*
'isexpand' 'ise'	string	(default: "")
			local to buffer
	Defines characters and patterns for completion in insert mode. Used by
	the |complete_match()| function to determine the starting position for
	completion. This is a comma-separated list of triggers. Each trigger
	can be:
	- A single character like "." or "/"
	- A sequence of characters like "->", "/*", or "/**"

> 	Note literal comma as a trigger character need using "\\,".

Reviewer 2: ```suggestion
	Note: use "\\," to have a literal comma in trigger characters.
```
